### PR TITLE
[6.13.z] Fix OEL Convert2Rhel tests

### PR DESCRIPTION
**Pull request by Satellite-QE**
Cherrypick of PR: https://github.com/SatelliteQE/robottelo/pull/14194

(cherry picked from commit 9f42080d00d9ac3fec70474cc3d276b4a5f8d07b)
Manual cherrypick of https://github.com/SatelliteQE/robottelo/pull/13236
### Problem Statement
OEL Convert2Rhel tests failing

### Solution
Use UBI repos required for subman packages, while registration instead of syncing.
Fix OEL tests by adding and modifying remediations for the inhibitors.
Replacing existing register_host() method with API method to generate register command - Fixes https://github.com/SatelliteQE/robottelo/issues/11876

### CODE REVIEW

**Comment by Satellite-QE** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py
pod_resources_size: extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5866
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_convert2rhel.py
Test Result : ================= 4 failed, 565 warnings in 4287.02s (1:11:27) =================
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py
pod_resources_size: extra_large

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5867
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_convert2rhel.py
Test Result : ================= 4 failed, 566 warnings in 4330.72s (1:12:10) =================
```

File: tests/foreman/api/test_convert2rhel.py [hunk 1]
```diff
@@ -276,7 +287,6 @@ def test_convert2rhel_oracle(module_target_sat, oracle, activation_key_rhel, ver
             'inputs': {
                 'Activation Key': activation_key_rhel.id,
                 'Restart': 'yes',
-                'Data telemetry': 'yes',
```
Comment (shubhamsg199) [inline tests/foreman/api/test_convert2rhel.py:279]: Removing this `Data telemetry': 'yes'` as it's not for 6.13.z

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5897
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_convert2rhel.py --external-logging
Test Result : ============ 3 passed, 769 warnings, 1 error in 6085.64s (1:41:25) =============
```

**Pull request merged**